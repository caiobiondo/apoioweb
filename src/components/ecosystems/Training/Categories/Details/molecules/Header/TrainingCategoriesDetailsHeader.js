import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { injectIntl, FormattedMessage } from 'react-intl';
import { Row, Col, getRowProps } from 'react-flexbox-grid';
import { Icon } from 'natura-ui';
import { Link } from 'react-router-dom';

import {
  Header,
  BackButtonWrapper,
  BackButtonIcon,
  BackButtonText,
  CategoryPercentageLabel,
  CategoryPercentageBar,
  BannerWrapper,
  Banner,
} from './TrainingCategoriesDetailsHeader.styles';

export class CategoryDetailsHeader extends Component {
  renderBanner = () => {
    const { category } = this.props;

    if (!category.banners || category.banners.length === 0) {
      return null;
    }

    const banner = category.banners[0];

    return <Banner thumbnail={banner.thumbnail} />;
  };

  getParsedPercentage = () => {
    const { category } = this.props;
    const { totalOfCoursesCompleted, totalOfCourses } = category;

    if (!totalOfCoursesCompleted || !totalOfCourses) {
      return 0;
    }

    return (totalOfCoursesCompleted * 100 / totalOfCourses).toFixed(2);
  };

  render() {
    const { category } = this.props;
    const rowProps = getRowProps({});
    const completedPercentage = this.getParsedPercentage();

    return (
      <Header>
        <Row>
          <Col md={4} sm={12}>
            <BackButtonWrapper>
              <Link to="/training/categories">
                <BackButtonIcon>
                  <Icon file="ico_back" />
                </BackButtonIcon>

                <BackButtonText>{category.name}</BackButtonText>
              </Link>
            </BackButtonWrapper>
          </Col>

          <Col md={8} sm={12}>
            <Row start="sm" end="md">
              <Col xs={12}>
                <CategoryPercentageLabel>
                  <strong>{completedPercentage}%</strong>&nbsp;
                  <FormattedMessage id="trainingCategoryCompleted" />
                </CategoryPercentageLabel>
                <CategoryPercentageBar completedPercentage={completedPercentage} />
              </Col>
            </Row>
          </Col>
        </Row>

        <BannerWrapper className={rowProps.className}>
          <Col xs={12}>{this.renderBanner()}</Col>
        </BannerWrapper>
      </Header>
    );
  }
}

CategoryDetailsHeader.propTypes = {
  category: PropTypes.object.isRequired,
};

export const CustomerDetailsWithIntl = injectIntl(CategoryDetailsHeader);

export default CustomerDetailsWithIntl;
