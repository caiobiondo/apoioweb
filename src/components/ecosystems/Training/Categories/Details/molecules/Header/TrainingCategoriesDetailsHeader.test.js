import React from 'react';
import { shallow } from 'enzyme';
import { CategoryDetailsHeader } from './TrainingCategoriesDetailsHeader';

const setup = propOverrides => {
  const props = Object.assign({}, propOverrides);

  const result = shallow(<CategoryDetailsHeader {...props} />);

  return {
    props,
    result,
  };
};

describe('Training Category Header', () => {
  it('render correct when there is at least 1 banner', () => {
    // given
    const category = {
      id: 1,
      name: 'Test',
      duration: 3600,
      totalOfCoursesCompleted: 5,
      totalOfCourses: 7,
      thumbnail: 'test.jpg',
      banners: [{ thumbnail: 'test.jpg' }],
    };

    // when
    const { result } = setup({ category });

    // then
    expect(result).toMatchSnapshot();
  });

  it('should return the correct percentage for a category', () => {
    // given
    const category = {
      id: 1,
      name: 'Test',
      duration: 3600,
      totalOfCoursesCompleted: 10,
      totalOfCourses: 20,
      thumbnail: 'test.jpg',
      banners: [{ thumbnail: 'test.jpg' }],
    };
    const expectedPercentage = '50';

    // when
    const { result } = setup({ category });
    const instance = result.instance();

    // then
    expect(instance.getParsedPercentage()).toEqual(expectedPercentage);
  });
});
