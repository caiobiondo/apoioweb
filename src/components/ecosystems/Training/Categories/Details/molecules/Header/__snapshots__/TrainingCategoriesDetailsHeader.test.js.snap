// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Training Category Header render correct when there is at least 1 banner 1`] = `
<TrainingCategoriesDetailsHeaderstyles__Header>
  <Row>
    <Col
      md={4}
      sm={12}
    >
      <TrainingCategoriesDetailsHeaderstyles__BackButtonWrapper>
        <Link
          replace={false}
          to="/training/categories"
        >
          <TrainingCategoriesDetailsHeaderstyles__BackButtonIcon>
            <Icon
              file="ico_back"
            />
          </TrainingCategoriesDetailsHeaderstyles__BackButtonIcon>
          <TrainingCategoriesDetailsHeaderstyles__BackButtonText>
            Test
          </TrainingCategoriesDetailsHeaderstyles__BackButtonText>
        </Link>
      </TrainingCategoriesDetailsHeaderstyles__BackButtonWrapper>
    </Col>
    <Col
      md={8}
      sm={12}
    >
      <Row
        end="md"
        start="sm"
      >
        <Col
          xs={12}
        >
          <TrainingCategoriesDetailsHeaderstyles__CategoryPercentageLabel>
            <strong>
              71
              %
            </strong>
            Â 
            <FormattedMessage
              id="trainingCategoryCompleted"
              values={Object {}}
            />
          </TrainingCategoriesDetailsHeaderstyles__CategoryPercentageLabel>
          <TrainingCategoriesDetailsHeaderstyles__CategoryPercentageBar
            completedPercentage="71"
          />
        </Col>
      </Row>
    </Col>
  </Row>
  <TrainingCategoriesDetailsHeaderstyles__BannerWrapper
    className="row"
  >
    <Col
      xs={12}
    >
      <TrainingCategoriesDetailsHeaderstyles__Banner
        thumbnail="test.jpg"
      />
    </Col>
  </TrainingCategoriesDetailsHeaderstyles__BannerWrapper>
</TrainingCategoriesDetailsHeaderstyles__Header>
`;
